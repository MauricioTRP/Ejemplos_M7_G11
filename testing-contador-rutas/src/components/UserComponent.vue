<template>
  <div>
    <h1>Lista de usuarios</h1>
    <ul>
      <li v-for="user in users" :key="user.id">{{ user.name }}</li>
    </ul>
  </div>
</template>
<script setup>
import { ref } from 'vue'

/**
 * m√°s logica para poblar la variable
 */
</script>
