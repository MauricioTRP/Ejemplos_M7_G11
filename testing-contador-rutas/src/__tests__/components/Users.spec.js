import { describe, it, expect, test } from 'vitest'
import { mount } from '@vue/test-utils'
import UserComponent from '@/components/UserComponent.vue'

describe('UserComponent', () => {
  /**
   * montar el componente
   * con variable de estado users
   */
  const wrapper = mount(UserComponent, {
    data() {
      return {
        users: [
          { id: 1, name: 'Juan Juan' },
          { id: 2, name: 'Juan Juan Juan' },
          { id: 3, name: 'Juan Juan Juan Juan' }
        ]
      }
    }
  })
  it('EL HTML se mantiene en las pruebas', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
  test('Tiene una lista de usuarios, con el texto "user.name"', () => {
    const li = wrapper.findAll('li')

    expect(li[0].text()).toBe('Juan Juan')
    expect(li).toHaveLength(3)
  })
})
