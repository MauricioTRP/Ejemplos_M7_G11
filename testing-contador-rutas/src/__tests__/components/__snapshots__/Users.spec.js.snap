// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserComponent > EL HTML se mantiene en las pruebas 1`] = `
"<div>
  <h1>Lista de usuarios</h1>
  <ul>
    <li>Juan Juan</li>
    <li>Juan Juan Juan</li>
    <li>Juan Juan Juan Juan</li>
  </ul>
</div>"
`;
